libs/utils/include/utils/FixedCapacityVector.h

```diff
@@ -46,6 +46,11 @@ namespace utils {
  * An empty vector with a given capacity is created with
  *   FixedCapacityVector<T>::with_capacity( capacity );
  *
+ * NOTE: One important difference betwen std::vector and FixedCapacityVector is that the former
```

Comment:
To perfectly correct, default construction *is* performed if an only if it is *not trivial*.  Or said differently, only trivial default construction is skipped.

Reply:
Good catch, I added a commit.